src/Containers/Pages/Metrics/Metrics.js

```diff
@@ -3,16 +3,44 @@ import MetricBtn from "../../../component/MetricBtn/MetricBtn";
 import messages from "../../../constant/messages";
 import BigRedBtn from "../../../component/BigRedBtn/BigRedBtn";
 import styles from "./Metrics.module.scss";
+import firebase from '../../../firebase';
+
 class Metrics extends Component {
+
   state = {
     selectedId: 1,
   };
 
+  updateMetric = (m) => {
+
+    var currentU = firebase.auth().currentUser
+      if (currentU == null)
+        currentU = "anonymous";
+      else
+        currentU = currentU = firebase.auth().currentUser.uid
+
+    firebase.database().ref('userID/' + currentU).set({
+      preference: {
+        metric: m
+      }
+    }
+    , (error) => {
+      if (error) {
+        // The write failed...
+        console.log("Write failed")
+      } else {
+        // Data saved successfully!
+        console.log("Write successful")
+      }
+    });
```

Comment:
this just a suggestion, it seems like a lot of these functions share similar code except the last portion where you have to set the path of the tree. How about unifying all of them into 1 place, and you can differentiate which one is getting called by sending in a 'type' variable as a parameter to the function.